widget_driver/test/drivable_widget_test.dart

```diff
@@ -66,5 +67,90 @@ void main() {
 
       expect(find.text('didCallTestMethod: true'), findsOneWidget);
     });
+    testWidgets('On state change updateDriverProvidedProperties gets called', (WidgetTester tester) async {
+      when(() => _mockRuntimeEnvironmentInfo.isRunningInTestEnvironment()).thenReturn(false);
+
+      final testContainerDrivableWidget = TestContainerDrivableWidget(
+        environmentInfo: _mockRuntimeEnvironmentInfo,
+      );
+      await tester.pumpWidget(MaterialApp(home: testContainerDrivableWidget));
+      final driver = testContainerDrivableWidget.driver;
+
+      expect(driver.updateCount, 1);
+
+      driver.aTestMethod();
+      await tester.pump();
+
+      expect(driver.updateCount, 2);
+    });
+
+    group('dispose behaviour of drivers', () {
+      testWidgets('does not create new driver, if there are no buildContext dependencies', (tester) async {
```

Comment:
Maybe we can have a chat about how to make these test cases more clear to the reader what we are testing here.